src/quick-lint-js/parse.h

```diff
@@ -1290,12 +1287,32 @@ class parser {
       QLJS_PARSER_UNIMPLEMENTED();
       break;
     }
-
+  }
+  template <QLJS_PARSE_VISITOR Visitor>
+  void parse_and_visit_function_body_no_scope(Visitor &v) {
     v.visit_enter_function_scope_body();
-
     this->parse_and_visit_statement_block_no_scope(v);
   }
 
+  template <QLJS_PARSE_VISITOR Visitor>
+  void parse_and_visit_function_parameters_and_body_no_scope(
+      Visitor &v, std::optional<source_code_span> name,
+      function_attributes attributes) {
+    function_guard guard = this->enter_function(attributes);
+    this->parse_and_visit_function_parameters_no_scope(v, name);
+    // function f() // Invalid.
+    if (this->peek().type != token_type::left_curly) {
+      if (!name.has_value()) {
+        QLJS_PARSER_UNIMPLEMENTED();
+      }
+      this->error_reporter_->report(error_missing_function_body{
+          .function_name = *name,
+      });
+      return;
+    }
+    this->parse_and_visit_function_body_no_scope(v);
```

Comment:
Nit: `no_scope` in the name `parse_and_visit_function_body_no_scope` is confusing. (It's also confusing for `parse_and_visit_function_parameters_and_body_no_scope`, but that's not your fault.) How about we inline the function to avoid the confusing function name entirely:
```suggestion
    v.visit_enter_function_scope_body();
    this->parse_and_visit_statement_block_no_scope(v);
```